code/datums/diseases/_MobProcs.dm

```diff
@@ -132,12 +134,13 @@
  *
  * Arguments:
  * * D - the disease the mob will try to contract
+ * * respect_carrier - if set to TRUE will not ignore the disease carrier flag
  */
 //Same as ContractDisease, except never overidden clothes checks
-/mob/proc/ForceContractDisease(datum/disease/D)
+/mob/proc/ForceContractDisease(datum/disease/D, respect_carrier)
 	if(!CanContractDisease(D))
 		return FALSE
-	AddDisease(D)
+	AddDisease(D, respect_carrier)
 	return TRUE
```

Comment:
I think you should add an optional parameter `notify_ghosts` which notifies ghosts when an infection takes place. This is some really nice behavior in the current implementation that I personally don't want to see disappear, and this seems like the best sort of place for it. 
```
    var/is_carrier = D.carrier == src
	for(var/mob/ghost as anything in GLOB.dead_mob_list) //Announce outbreak to dchat
		to_chat(ghost, "<span class='deadsay'><b>Disease outbreak:</b>[character] ([ghost_follow_link(character, ghost)]) [is_carrier ? "is now a carrier of" : "has contracted"] [GLOB.current_pending_disease]!</span>")
```

Reply:
But I'm already notifying ghosts in the event start() itself?